src/OrchardCore.Modules/OrchardCore.Users/Controllers/AdminController.cs

```diff
@@ -528,13 +529,25 @@ public async Task<IActionResult> EditPassword(ResetPasswordViewModel model)
 
         if (ModelState.IsValid)
         {
-            var token = await _userManager.GeneratePasswordResetTokenAsync(user);
+            if (user.UserName == model.UsernameOrEmail)
```

Comment:
This logic should not be needed if we are asking the current user to provide their password always

Reply:
This is useful to distinguish between chaning & resetting password

Reply:
This action is for changing the password not resetting. Resettings password is done using the frontend UI.

Changing the password is an admin or privilege user change the password for other users "or theme self". I think we should ask the user for their current password always to prevent a user from change passwords for user "like if a user left their PC unlocked or something like that".

Reply:
If the user edits his password, the code should call `ChangePasswordAsync()`; otherwise, `ResetPasswordAsync()` regardless if it's call from site or admin

> I think we should ask the user for their current password always to prevent a user from change passwords for user "like if a user left their PC unlocked or something like that".

Admins & site owners should be able to edit the password for other users (Resetting in their behalves)

Reply:
You can always use ResetPasswordAsync but confirm the password first. So this way we know the admin provided their password and it is valid. And if it valid, reset it